shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Validation.hs

```diff
@@ -92,11 +91,12 @@ applyTickTransition ::
   SlotNo ->
   ShelleyState crypto
 applyTickTransition globals state hdr =
-  fromRight err . flip runReader globals
+  (either err identity) . flip runReader globals
     . applySTS @(STS.TICK crypto)
     $ TRC (mkTickEnv state, state, hdr)
   where
-    err = error "Panic! applyHeaderTransition failed."
+    err :: Show a => a -> LedgerState.NewEpochState crypto
```

Comment:
Maybe nicer to just have `b` on the RHS, since that emphasizes the error nature of this.